### PR TITLE
dird: fix director resource not showing when using `show director` or the `--xc director` cli option

### PR DESCRIPTION
**Pull request by alaaeddineelamri**
#### Description

`show directors` command does not work as expected due to a previous update to enums.
This PR fixes the issue.

#### Please check

- [x] Short description and the purpose of this PR is present _above this paragraph_
- [x] Your name is present in the AUTHORS file (optional)

If you have any questions or problems, please give a comment in the PR.

### Helpful documentation and best practices

- [Git Workflow](https://docs.bareos.org/DeveloperGuide/gitworkflow.html)
- [Automatic Sourcecode Formatting](https://docs.bareos.org/DeveloperGuide/generaldevel.html#automatic-sourcecode-formatting)
- [Check your commit messages](https://docs.bareos.org/DeveloperGuide/gitworkflow.html#commits)

##### General
- [x] PR name is meaningful
- [x] Purpose of the PR is understood
- [x] Commit descriptions are understandable and well formatted
- [x] Check backport line
- [x] Is the PR title usable as CHANGELOG entry?
- [x] Separate commit for CHANGELOG.md ("update CHANGELOG.md"). The PR number is correct.

##### Source code quality

- [x] Source code changes are understandable
- [x] Variable and function names are meaningful
- [x] Code comments are correct (logically and spelling)
- [x] Required documentation changes are present and part of the PR
- [x] `bareos-check-sources --since-merge` does not report any problems

##### Tests

- [x] Decision taken that a test is required (if not, then remove this paragraph)
- [x] The choice of the type of test (unit test or systemtest) is reasonable
- [x] Testname matches exactly what is being tested
- [x] On a fail, output of the test leads quickly to the origin of the fault

### CODE REVIEW

File: core/src/dird/ua_output.cc [hunk 1]
```diff
@@ -254,67 +260,57 @@ bool show_cmd(UaContext* ua, const char*)
...
+          if (!res) { type = -2; }
           break;
         }
       }
     }
 
     switch (type) {
-      case -1: /* all */
-        ShowAll(ua, hide_sensitive_data, verbose);
+      case -1:
```
Comment (joergsteffens) [inline core/src/dird/ua_output.cc:301]: Can you define these magic numbers somewhere?

**Review by joergsteffens — CHANGES_REQUESTED** [review]
Testing shows, that the bconsole show command works as expected.
However, there are (now?) discrepancies between the bconsole show command and the --export-config command line argument:

### bconsole:
```
*show DONTEXIST
Resource DONTEXIST not found

*show job=DONTEXIST
job resource DONTEXIST not found.
```
Okay.

### cmdline:

```
sbin/bareos_dir-python-bareos  -c etc/bareos --xc job DONTEXIST
No ***UNKNOWN*** resource defined

$ sbin/bareos_dir-python-bareos  -c etc/bareos --xc director
<no outputp>

$ sbin/bareos_dir-python-bareos  -c etc/bareos --xc DONTEXIST
<no outputp>
```

`--xc director` doesn't work. Missing error output.

### Other differences:
* bconsole:
  * show job => show jobdefs
  * show jobs => show jobs
* cmd:
  * `--xc job => show jobs`
  * `--xc jobs => <empty>`

The job and jobs keywords are handled differently.

**Comment by joergsteffens** [comment]
Please also run `bareos-check-sources` and maybe you can find a  better usable title to also be addable to the CHANGELOG file.

**Comment by alaaeddineelamri** [comment]
The differences between show and export config were already there on master, so the changes here did not add anything new to that.

The issue with
```
$ sbin/bareos_dir-python-bareos  -c etc/bareos --xc director
<no outputp>
```
is the same thing in relation to the enum changes, but not directly involved with the show command. I pushed a fix for that too.

The show command and the export config option do not use exactly the same code. In case we want them to behave the same, I can implement changes for that. Otherwise, if we want to keep this PR uniquely for the purpose of fixing the director config not showing up, we can do the unification of behavior in another PR.

**Review by joergsteffens — APPROVED** [review]

**Pull request merged**